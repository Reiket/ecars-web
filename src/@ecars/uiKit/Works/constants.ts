import {withBlockClass} from 'ecars-web-lib';
import {SectionWrapper} from '@ecars/uiKit/Section/components/SectionWrapper';

export const WorksWrapperHOC = withBlockClass(SectionWrapper, 'works');

export interface WorksStepType {
  title: string;
  text: string;
  id: number;
}

export const worksStepConfig: WorksStepType[] = [
  {
    id: 1,
    title: 'Find a car',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.',
  },
  {
    id: 2,
    title: 'Extensive inspection',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.',
  },
  {
    id: 3,
    title: 'Safe Buying',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.',
  },
  {
    id: 4,
    title: 'Delivery and support',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.',
  },
];
