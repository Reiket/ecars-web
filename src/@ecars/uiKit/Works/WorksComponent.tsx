import type {FC} from 'react';
import {Works} from '@ecars/uiKit/Works';

export const WorksComponent: FC = () => (
  <Works.Wrapper name="works">
    <div className="works__body">
      <h2 className="works__title section-title">How it works</h2>
      <p className="works__text">
        Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim
        velit mollit.
      </p>
      <Works.List />
    </div>
  </Works.Wrapper>
);
