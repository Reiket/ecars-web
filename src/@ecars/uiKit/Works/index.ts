import {WorksComponent} from '@ecars/uiKit/Works/WorksComponent';
import {WorksWrapperHOC} from '@ecars/uiKit/Works/constants';
import {WorksStep} from '@ecars/uiKit/Works/components/WorksStep';
import {WorksList} from '@ecars/uiKit/Works/components/WorksList';

export const Works = Object.assign(WorksComponent, {
  Wrapper: WorksWrapperHOC,
  Step: WorksStep,
  List: WorksList,
});
