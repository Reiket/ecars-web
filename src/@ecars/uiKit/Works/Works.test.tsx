import {render} from '@testing-library/react';
import {Works} from '@ecars/uiKit/Works/index';

describe('Works Component', () => {
  test('render component correctly', () => {
    const {container} = render(<Works />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
