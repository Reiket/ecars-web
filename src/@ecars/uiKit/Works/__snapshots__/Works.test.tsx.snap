// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Works Component > render component correctly 1`] = `
<div>
  <div
    class="works"
  >
    <section
      class="works__section section"
    >
      <div
        class="works__container container container--lg"
      >
        <div
          class="works__body"
        >
          <h2
            class="works__title section-title"
          >
            How it works
          </h2>
          <p
            class="works__text"
          >
            Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit.
          </p>
          <ol
            class="works__works-list works-list"
          >
            <li
              class="works-list__item"
            >
              <span
                class="works-list__number"
              >
                1
              </span>
              <h4
                class="works-list__title title-works-list"
              >
                Find a car
              </h4>
              <p
                class="works-list__text"
              >
                Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
              </p>
            </li>
            <li
              class="works-list__item"
            >
              <span
                class="works-list__number"
              >
                2
              </span>
              <h4
                class="works-list__title title-works-list"
              >
                Extensive inspection
              </h4>
              <p
                class="works-list__text"
              >
                Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
              </p>
            </li>
            <li
              class="works-list__item"
            >
              <span
                class="works-list__number"
              >
                3
              </span>
              <h4
                class="works-list__title title-works-list"
              >
                Safe Buying
              </h4>
              <p
                class="works-list__text"
              >
                Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
              </p>
            </li>
            <li
              class="works-list__item"
            >
              <span
                class="works-list__number"
              >
                4
              </span>
              <h4
                class="works-list__title title-works-list"
              >
                Delivery and support
              </h4>
              <p
                class="works-list__text"
              >
                Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
              </p>
            </li>
          </ol>
        </div>
      </div>
    </section>
  </div>
</div>
`;
